forecasttools/data.py

```diff
@@ -84,8 +84,12 @@ def make_census_dataset(
             pl.col("STATE_NAME").alias("long_name"),
         ]
     )
-    location_table = nation.vstack(jurisdictions)
-    location_table.write_csv(file_save_path)
+    location_table = nation.vstack(jurisdictions).with_columns(
+        pl.col("location_code").map_elements(
+            lambda x: f"0{x}" if len(x) == 1 else x, return_dtype=pl.Utf8
+        )
```

Comment:
(1) `read_parquet` is being used in `__init__`, (2) `schema_overrides` is a non-existent parameter of `read_parquet`, (3) regarding `schema` (see [here]()), we have _Specify the datatypes of the columns. The datatypes must match the datatypes in the file(s). If there are extra columns that are not in the file(s), consider also enabling `allow_missing_columns`._ with the Warning: _<span style="color: red;">This functionality is considered unstable. It may be changed at any point without it being considered a breaking change</span>._, (4) `location_code` already includes "US" and is read in as `pl.Utf8`, so stating this again with `schema = {"location_code": pl.Utf8}` doesn't seem valuable



What I intend to do is use the following in `__init__.py`:

Change

```python
# location table (from Census data)
with importlib.resources.path(
    __package__, "location_table.parquet"
) as data_path:
    location_table = pl.read_parquet(data_path)
```

to

```python
# location table (from Census data)
with importlib.resources.path(
    __package__, "location_table.parquet"
) as data_path:
    location_table = pl.read_parquet(data_path).with_columns(
        pl.col("location_code").str.zfill(2)
    )
```

Reply:
This isn't a comment on the read from parquet, it's a comment on the code that constructs the dataset.

Reply:
Ah, I see. I interpreted the `read_csv()` as in "where the location table data object (parquet)" is read in. 

In light of this, what would you prefer the suggested edit to be? (I am looking at `schema` and `schema_overrides` and do not see an immediate way to do something akin to `pl.col("location_code").str.zfill(2)`).

Reply:
The source data has zero padding, so if you just read the column as a string column it will be fine.

Reply:
The source data does not have 0-padding. But it needs this, at least for the FluSight Hub. By default, the `location_code` is read in as `pl.Utf8`. 1 comes up as "1". However, the FluSight submissions seem to require code "1" to actually be "01", and so on. The example FluSight submission has zero padding and Polars dataframe for `location_code` (just reading in using `read_csv`) does not pad with zeroes by default.

Reply:
I see zero padding: https://www2.census.gov/geo/docs/reference/state.txt

<img width="372" alt="Screenshot 2024-11-05 at 1 52 35 PM" src="https://github.com/user-attachments/assets/ac2b0a1f-6327-47aa-8d2b-59403f239867">

Reply:
That's so odd. 

In testing, the following returns "1" for AL.

`pl.col("STATE").alias("location_code").cast(pl.Utf8)` 

(I've had this for a long time unchanged) .

Reply:
Let me check over things again.

Reply:
This is the point of my comment. At the `pl.read_csv` stage, the column is parsed as an `int`, because it contains only (zero-padded) ints. But if you explicitly tell `pl.read_csv` to parse it as a string, you will get `"01", "02",` etc